client/components/contact/ContactForm.jsx

```diff
@@ -0,0 +1,270 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { sendGitRequest } from '@reducers/data';
+import { showFeedbackSuccess, setErrorModal } from '@reducers/ui';
+import { toast } from 'react-toastify';
+
+import 'react-toastify/dist/ReactToastify.css';
+
+import {
+  Container,
+  Grid,
+  Button,
+  TextField,
+  CircularProgress,
+} from '@material-ui/core';
+
+const initialFormValues = {
+  firstName: '',
+  lastName: '',
+  email: '',
+  association: '',
+  message: '',
+  errors: {
+    missingFirstName: false,
+    missingLastName: false,
+    missingEmail: false,
+    invalidEmail: false,
+    missingMessage: false,
+  },
+  loading: false,
+};
+
+const ContactForm = () => {
+  // define the methods to dispatch redux actions
+  const dispatch = useDispatch();
+  const callSendGitRequest = useCallback(obj => dispatch(sendGitRequest(obj)), [dispatch]);
```

Comment:
eslint prevents lines > 100 characters so i had to be terse lol. also a bit difficult to provide anything more meaningful since payload is an generic object until it is used in a given context. i'm open to any ideas tho

Reply:
Hmm based on what I'm seeing, they are being used in specific contexts here.

line 35: obj -> gitIssueContent
line 36: o -> isSuccessful
line 37: o -> hasError

For the purpose of readability, it's better to be a little more verbose and split a line into two rather than trying to keep one line less than 100 chars. Breaking lines in js: https://stackoverflow.com/a/35167886

Reply:
Eslint is playing games with me here. I tried a few ways to break up the lamba functions but eslint keeps recreating the one-liner and when I change the variable name, i'm exceeding 100 chars lol. Should we eslint ignore these lines?

Reply:
updated as discussed by calling dispatch directly in respective locations instead of defining a lambda